File: Dihya/ai/test_ai.js
// Tests unitaires pour Dihya AI (Node.js)
// Couvre la génération, la sécurité, le fallback, la multilingue, la conformité RGPD.
const { generateAIResponse, selectEngine, SUPPORTED_LANGUAGES } = require('./ai');
test('Génération IA FR', async () => {
  const result = await generateAIResponse({ prompt: 'Bonjour', userId: 'test', lang: 'fr', role: 'user', engine: 'mixtral' });
  expect(result).toMatch(/Bonjour/);
});
test('Fallback engine', async () => {
  const engine = await selectEngine('openai');
  expect(['openai', 'mixtral', 'llama', 'mistral']).toContain(engine);
});
test('Langues supportées', () => {
  expect(SUPPORTED_LANGUAGES).toEqual(expect.arrayContaining(['fr', 'en', 'ar', 'tzr']));
});

File: Dihya/ai/test_ai.py
"""
Tests unitaires et d'intégration pour le module IA Dihya (Python)
Couvre la génération, la sécurité, le fallback, la multilingue, la conformité RGPD.
"""
import unittest
from ai import generate_ai_response, select_engine, SUPPORTED_LANGUAGES

class TestDihyaAI(unittest.TestCase):
    def test_generate_fr(self):
        result = generate_ai_response(prompt="Bonjour", user_id="test", lang="fr", role="user", engine="mixtral")
        self.assertIn("Bonjour", result)
    def test_fallback_engine(self):
        engine = select_engine(preferred="openai")
        self.assertIn(engine, ["openai", "mixtral", "llama", "mistral"])
    def test_supported_languages(self):
        self.assertIn("fr", SUPPORTED_LANGUAGES)
        self.assertIn("en", SUPPORTED_LANGUAGES)
        self.assertIn("ar", SUPPORTED_LANGUAGES)
        self.assertIn("tzr", SUPPORTED_LANGUAGES)
if __name__ == "__main__":
    unittest.main()
